### PR TITLE
docs: Update version-and-publish.md

### PR DESCRIPTION
**Pull request by Civon**
Is this typo? 
Since the example app does mark private in package.json

ref: https://github.com/lerna/getting-started-example/blob/2e37c43fbbd4f1fc050a766d6585f88d4ef3f79d/packages/remixapp/package.json#L4C19-L4C19

## Description
Match example repo's package private status with doc.

## Motivation and Context
May fix paradox between two doc 

## How Has This Been Tested?
N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ x] Chore (change that has absolutely no effect on users)

## Checklist:

- [ ] My code follows the code style of this project.
- [ x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: website/docs/features/version-and-publish.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ The repo contains three packages or projects:
 We are going to publish the `header` and the `footer` packages.
 
 > It's common to publish only a subset of the projects. Some projects can be private (e.g., used only for tests), some
-> can be demo apps. In this repo, `remixapp` isn't private, it just doesn't get published to NPM.
+> can be demo apps. In this repo, `remixapp` is private, it just doesn't get published to NPM.
```
Comment (JamesHenry) [inline website/docs/features/version-and-publish.md:24]:
```suggestion
> can be demo apps. In this repo, `remixapp` isn't "conceptually private", it is just using the `"private": true` setting in order to not get published to NPM.
```

**Comment by JamesHenry** [comment]
@Civon It isn't a typo, but you are not the first person who has been confused by this, so I have updated your PR to clarify it

**Comment by Civon** [comment]
Thanks for the clarification 🤗 

If the "conceptually private" means that the repo is private or not. The sentence below is easier understand for me.

```
> It's common to publish only a subset of the projects. Some projects can be private (e.g., used only for tests), some
> can be demo apps. In this example, the `remixapp` uses the `"private": true` setting to prevent it from being published to NPM.
```

File: website/docs/features/version-and-publish.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ The repo contains three packages or projects:
 We are going to publish the `header` and the `footer` packages.
 
 > It's common to publish only a subset of the projects. Some projects can be private (e.g., used only for tests), some
-> can be demo apps. In this repo, `remixapp` isn't private, it just doesn't get published to NPM.
+> can be demo apps. In this repo, `remixapp` isn't "conceptually private", it is just using the `"private": true` setting in order to not get published to NPM.
```
Comment (JamesHenry) [inline website/docs/features/version-and-publish.md:24]:
```suggestion
> can be demo apps. In this repo, `remixapp` isn't "private" in the sense of not wanting people to see the source files, it is just using the `"private": true` setting in order to not get published to NPM.
```

**Comment by JamesHenry** [comment]
@Civon I have updated it to further simplify the explanation

**Pull request merged**